src/charm.py

```diff
@@ -79,10 +106,49 @@ def _update_pebble_layer(self, event: HookEvent) -> None:
     def _on_ingress_ready(self, event: IngressPerAppReadyEvent) -> None:
         if self.unit.is_leader():
             logger.info("This app's public ingress URL: %s", event.url)
+        self._update_pebble_layer(event)
+        self._update_login_ui_endpoint_relation_data(event)
 
     def _on_ingress_revoked(self, event: IngressPerAppRevokedEvent) -> None:
         if self.unit.is_leader():
             logger.info("This app no longer has ingress")
+        self._update_pebble_layer(event)
+        self._update_login_ui_endpoint_relation_data(event)
+
+    def _update_login_ui_endpoint_relation_data(self, event: RelationEvent) -> None:
+        endpoint = (
+            self.ingress.url
+            if self.ingress.is_ready()
+            else f"{self.app.name}.{self.model.name}.svc.cluster.local:{APPLICATION_PORT}",
```

Comment:
I would suggest we provide no endpoint if ingress is not ready. The login ui needs to be accessible from the user and this domain name makes sense only within the k8s cluster.